### PR TITLE
Various cleanups and removals of unused code.

### PR DESCRIPTION
**Pull request by bigkevmcd**
**What type of PR is this?**

/kind cleanup

**What does this PR do / why we need it**:

A lot of linter-driven and review-driven changes to tidy up a lot of code, removing unused parameters, functions, unnecessary declarations.

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: pkg/pipelines/argocd/argocd.go [hunk 1]
```diff
@@ -60,17 +60,21 @@ type resource struct {
 }
 
 const (
+	// ArgoCDNamespace is the default namespace for ArgoCD installations.
+	ArgoCDNamespace = "argocd"
+
```
Comment (wtam2018) [inline pkg/pipelines/argocd/argocd.go:65]: indentations don't line up.   run golint?

File: pkg/pipelines/bootstrap.go [hunk 2]
```diff
@@ -45,9 +45,9 @@ const (
 	rolesPath             = "02-rolebindings/pipeline-service-role.yaml"
 	rolebindingsPath      = "02-rolebindings/pipeline-service-rolebinding.yaml"
 	serviceAccountPath    = "02-rolebindings/pipeline-service-account.yaml"
-	secretsPath           = "03-secrets/gitops-webhook-secret.yaml"
-	authTokenPath         = "03-secrets/git-host-access-token.yaml"
-	basicAuthTokenPath    = "03-secrets/git-host-basic-auth-token.yaml"
+	secretsPath           = "03-secrets/gitops-webhook-secret.yaml"     //nolint:gosec
+	authTokenPath         = "03-secrets/git-host-access-token.yaml"     // nolint:gosec
+	basicAuthTokenPath    = "03-secrets/git-host-basic-auth-token.yaml" // nolint:gosec
```
Comment (wtam2018) [inline pkg/pipelines/bootstrap.go:50]: If there are gosec warnings, can they be fixed?

File: pkg/pipelines/bootstrap.go [hunk 3]
```diff
@@ -405,29 +398,29 @@ func checkPipelinesFileExists(appFs afero.Fs, outputPath string, overWrite bool)
 func createInitialFiles(fs afero.Fs, repo scm.Repository, o *BootstrapOptions) (res.Resources, error) {
 	cicd := &config.PipelinesConfig{Name: o.Prefix + "cicd"}
 	pipelineConfig := &config.Config{Pipelines: cicd}
-	pipelines := createManifest(repo.URL(), pipelineConfig)
+	manifest := createManifest(repo.URL(), pipelineConfig)
 	initialFiles := res.Resources{
-		pipelinesFile: pipelines,
+		pipelinesFile: manifest,
```
Comment (wtam2018) [inline pkg/pipelines/bootstrap.go:403]: We changes between pipelines <-> manifest many times.  (Sigh)

File: pkg/pipelines/bootstrap_test.go [hunk 4]
```diff
@@ -36,7 +36,7 @@ func TestBootstrapManifest(t *testing.T) {
 	}(secrets.DefaultPublicKeyFunc)
 
 	secrets.DefaultPublicKeyFunc = func(service types.NamespacedName) (*rsa.PublicKey, error) {
-		key, err := rsa.GenerateKey(rand.Reader, 1024)
+		key, err := rsa.GenerateKey(rand.Reader, 2048)
```
Comment (wtam2018) [inline pkg/pipelines/bootstrap_test.go:39]: Make it a const?

File: pkg/pipelines/config/validate.go [hunk 5]
```diff
@@ -12,7 +12,8 @@ import (
 )
 
 const (
-	LongServiceNameError = "A service name can be no longer than 47 characters"
+	longServiceName  = "A service name can be no longer than 47 characters"
```
Comment (wtam2018) [inline pkg/pipelines/config/validate.go:15]: Consider:
"A service name cannot exceed 47 characters."?

File: pkg/pipelines/argocd/argocd.go (hunk 1 above)
Comment (bigkevmcd) [reply pkg/pipelines/argocd/argocd.go:65]: This is actually the behaviour of gofmt, each group shares the same indentation.

I split the exported `ArgoCDNamespace` from the internal names, and so each group is indented separately.

File: pkg/pipelines/bootstrap.go (hunk 2 above)
Comment (bigkevmcd) [reply pkg/pipelines/bootstrap.go:50]: I think gosec finds things like `token` and `secret` in names.

Alternative name suggestions that don't include `secret` or `token` welcome...

File: pkg/pipelines/bootstrap.go (hunk 3 above)
Comment (bigkevmcd) [reply pkg/pipelines/bootstrap.go:403]: Yeah, the result of calling `createManifest` is stored in `manifest` and we have `manifestFile`.

I suspect we could take another pass at naming here.

File: pkg/pipelines/bootstrap_test.go (hunk 4 above)
Comment (bigkevmcd) [reply pkg/pipelines/bootstrap_test.go]: I did something slightly nicer (removed the repetition).

**Review by wtam2018 — APPROVED** [review]

**Pull request merged**